src/Sarif.Converters/PyLintConverter.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis.Sarif.Writers;
+using Microsoft.CodeAnalysis.Sarif.Converters.PyLintObjectModel;
+
+namespace Microsoft.CodeAnalysis.Sarif.Converters
+{
+    public class PyLintConverter : ToolFileConverterBase
+    {
+        public override void Convert(Stream input, IResultLogWriter output, LoggingOptions loggingOptions)
+        {
+            input = input ?? throw new ArgumentNullException(nameof(input));
+            output = output ?? throw new ArgumentNullException(nameof(output));
+
+            PyLintLogList pyLintDefects = PyLintLogReader.ReadLog(input);
+
+            Tool tool = new Tool
+            {
+                Name = "PyLint",
+                FullName = "PyLint Static Code Analysis"
+            };
+
+            output.Initialize(id: null, automationId: null);
+            output.WriteTool(tool);
+
+            var results = new List<Result>();
+
+            foreach (PyLintLog entry in pyLintDefects)
+            {
+                results.Add(this.PrepareSarifLogEntry(entry));
+            }
+
+            var fileInfoFactory = new FileInfoFactory(MimeType.DetermineFromFileExtension, loggingOptions);
+            Dictionary<string, FileData> fileDictionary = fileInfoFactory.Create(results);
+
+            if (fileDictionary?.Any() == true)
+            {
+                output.WriteFiles(fileDictionary);
+            }
+
+            output.OpenResults();
+            output.WriteResults(results);
+            output.CloseResults();
+        }
+
+        internal Result PrepareSarifLogEntry(PyLintLog defect)
```

Comment:
The `TSLint` converter uses the name `CreateResult` for this -- I think that is a better name.